pkg/gameserverallocations/controller.go

```diff
@@ -137,11 +168,22 @@ func (c *Controller) Run(workers int, stop <-chan struct{}) error {
 	c.stop = stop
 
 	c.logger.Info("Wait for cache sync")
-	if !cache.WaitForCacheSync(stop, c.gameServerAllocationSynced) {
+	if !cache.WaitForCacheSync(stop, c.gameServerAllocationSynced, c.gameServerSynced) {
 		return errors.New("failed to wait for caches to sync")
 	}
 
+	// build the cache
+	err = c.syncReadyGSServerCache()
+	if err != nil {
+		return err
+	}
+
 	c.workerqueue.Run(workers, stop)
+
+	// we don't want mutiple workers refresh cache at the same time so one worker will be better.
+	// Also we don't expect to have too many failures when allocating
+	c.gsWorkerqueue.Run(1, stop)
```

Comment:
how can it be? Run() is a blocking call.

Reply:
If we have more than one worker, each of them consume messages from the queue (in parallel) which will effectively try to refresh the cache (the refresh method refreshes the whole cache)